File: benchmarks/gcd/gcd.c
#include <stdio.h>

int gcd(int a, int b) {
    while (b != 0) {
        int t = b;
        b = a % b;
        a = t;
    }
    return a;
}

int main() {
    int result = 0;
    int i = 1;
    while (i <= 22000000) {
        result += gcd(i, 31415927);
        i += 1;
    }
    printf("%d\n", result);
    return 0;
}

File: benchmarks/gcd/gcd.go
package main

import "fmt"

func gcd(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func main() {
	result := 0
	i := 1
	for i <= 22000000 {
		result += gcd(i, 31415927)
		i++
	}
	fmt.Println(result)
}
